NEW_COMPONENT.md

```diff
@@ -8,80 +8,159 @@ This repository contains two separate projects - the component library and the d
+@Component({
+    selector: 'app-poster',
+    templateUrl: './poster-docs.component.html'
+})
+export class PosterDocsComponent {
 
-Now, the <fd-poster> component will be available for use by the documentation application.
+    posterHtml = posterHtml;
 
-## Add an image to the Poster component template
+}
+```
 
-In the library source, open `poster.component.html`, remove the default code and add an image with a placeholder.
+Note that we're using raw-loader to import the poster example code as raw text.  This text will be rendered as the example source.
 
-```<img src="https://placeimg.com/400/600/nature'"/>```
+In `poster-docs.component.html`, we'll provide a brief explanation of the poster component, and we'll add the poster component itself, along with the code example.
+
+```
+<h2>Poster</h2>
+<description>
+    The Poster component displays a photograph taken from an image placeholder site.
+</description>
+<component-example [name]="'ex1'">
+    <app-poster-example></app-poster-example>
+</component-example>
+<code-example [code]="posterHtml" [language]="'HTML'"></code-example>
+```
 
-## Add the Poster component to the Poster Docs template
+## Adding the new documentation module and route
 
-In the documentation project, open `poster-docs.component.html`, remove the default code and add the new poster component:
+Now that we've got our documentation files for the poster, add them to the documentation module declarations array in `documentation.module.ts`.
 
-```<fd-poster></fd-poster>```
+```TypeScript
+    declarations: [
+        PosterDocsComponent,
+        PosterExampleComponent,
+```
 
-Now, when you navigate to the Poster docs through the side navigation, you'll see our new Poster component!
+Be sure to import these at the top of the file as well.
+
+We use TypeDoc to automatically generate TypeScript documentation for explanations of inputs, outputs, etc.  We won't go in to details on TypeDoc in this tutorial, but know that all files we wish to have TypeDocs for must be referenced in `docs/app/documentation/utilities/api-files.ts`.  Open the file and add the following to the `API_FILES` object: 
+
+```TypeScript
+poster: [
+    'PosterComponent'
+],
+```
+
+Let's add a poster route, and put a link for the new docs in the 'Components' side bar.
+
+Open `documentation.routes.ts` and add the following to the `children` in the `ROUTES` array:
+
+```TypeScript
+{ path: 'poster', component: PosterComponent, children: [
+        { path: '', component: PosterDocsComponent},
+        { path: 'api', component: ApiComponent, data: {content: API_FILES.poster}}
+    ]
+},
+```
+You will need to import `PosterComponent` and `PosterDocsComponent` in this file as well.
+
+Next, open `documentation.component.ts` and add the following to the `components` array:
 
-![alt text](assets/images/new_component/finished-poster-component.png "Finished Poster Component")
+```javascript
+{ url: 'poster', name: 'Poster' }
```

Comment:
I think json is an option for these code blocks

Reply:
keys must be strings in json, which is not what we're using for the rest of these objects